packages/s3-store/index.ts

```diff
@@ -422,7 +423,12 @@ export class S3Store extends DataStore {
     this.cache.delete(id)
   }
 
-  private calcOptimalPartSize(size: number): number {
+  private calcOptimalPartSize(size?: number): number {
+    // When upload size is not know we assume largest possible value (`maxUploadSize`)
+    if (size === undefined) {
+      size = this.maxUploadSize
```

Comment:
```suggestion
    // When upload size is not know we assume largest possible value
    if (size === undefined) {
      size = this.maxPartSize
```

Reply:
Variable `size` is total size of upload (`Upload-Length`), which can be deferred and thus undefined. In this case we can not assume much, but upper limit is still 5TB (max upload size). Using this `size` this function will calculate appropriate part size while also considering max part size and maximum number of parts.

This logic has some flaws but it works for general case and does not exceed max part size.

Reply:
ah of course. I often glance too quickly (with dyslexia) and managed to think we were ending up with the part size of `maxUploadSize`. All good here 👍